vmngclient/api/template_api.py

```diff
@@ -56,50 +54,129 @@ def __init__(self, name):
+        self, summary: bool = True, offset: Optional[int] = None, limit: Optional[int] = None
+    ) -> DataSequence[FeatureTemplateInfo]:
+        """In a multitenant vManage system, this API is only available in the Provider view."""
+        endpoint = "/dataservice/template/feature"
+        params = {"summary": summary}
 
-        Returns:
-            List[Template]: List of existing templates.
-        """
+        fr_templates = self.session.get(url=endpoint, params=params)
+
+        return fr_templates.dataseq(FeatureTemplateInfo)
+
+    def _get_device_templates(
+        self, feature: DeviceTemplateFeature = DeviceTemplateFeature.ALL
+    ) -> DataSequence[DeviceTemplateInfo]:
+        """In a multitenant vManage system, this API is only available in the Provider view."""
         endpoint = "/dataservice/template/device"
-        data = self.session.get_data(endpoint)
-        return [create_dataclass(Template, template) for template in data]
+        params = {"feature": feature.value}
 
-    def get(self, name: str) -> Template:
-        """
+        templates = self.session.get(url=endpoint, params=params)
+        return templates.dataseq(DeviceTemplateInfo)
 
-        Args:
-            name (str): Name of template.
+    @overload
+    def get(self, template: type) -> DataSequence[TemplateInfo]:
+        ...
 
-        Raises:
-            TemplateNotFoundError: If template does not exist.
+    @overload
+    def get(self, template: DeviceTemplate) -> DataSequence[TemplateInfo]:
+        ...
 
-        Returns:
-            Template: Selected template.
-        """
-        for template in self.templates:
-            if name == template.name:
-                return template
-        raise TemplateNotFoundError(name)
+    def get(self, template):
+        if template is FeatureTemplate:
+            return self._get_feature_templates()
 
-    def get_id(self, name: str) -> str:
-        """
+        if template is DeviceTemplate:
+            return self._get_device_templates()
 
-        Args:
-            name (str): Name of template to get id.
+        return DataSequence(TemplateInfo)
 
-        Returns:
-            str: Template id.
-        """
-        return self.get(name).id
+    @overload
+    def attach(self, template: CLITemplate, name: str, device: Device) -> bool:
+        ...
+
+    @overload
+    def attach(self, template: DeviceTemplate) -> bool:
+        ...
+
+    def attach(self, template, device, name=None, **kwargs):
+        if isinstance(template, CLITemplate):
+            return self._attach_cli(template, name, device)
+
+        if isinstance(template, DeviceTemplate):
+            return self.attach_feature(template.name, device, **kwargs)
+
+        if template is DeviceTemplate and name:
+            self.attach_feature(name, device, **kwargs)
+
+        raise NotImplementedError()
+
+    # TODO list of devices
+    def attach_feature(self, name: str, device: Device, **kwargs):
```

Comment:
as `**kwargs` can be important for the user, I would ask to add docstring explaining why they should be dict that specifies`"device_specific_vars"`. Not needed if this is mentioned in documentation on wiki